### PR TITLE
arch/risc-v: Refine riscv_assert.c and arch.h

### PR DESCRIPTION
**Pull request by no1wudi**

### CODE REVIEW

File: arch/risc-v/src/common/riscv_assert.c [hunk 1]
```diff
@@ -57,6 +57,14 @@
 #  define CONFIG_BOARD_RESET_ON_ASSERT 0
 #endif
 
+/* Format output with register width and hex */
+
+#ifdef CONFIG_ARCH_RV32
+#  define PRIxREG "8"PRIx32
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_assert.c:63]: let's use PRIxPTR to avoid chceck CONFIG_ARCH_RV32

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_assert.c:63]: Yes, but we still need to specific the output width for rv32 (8) and rv64 (16) by it.

Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_assert.c:63]: yes, but we can check __LP64__ like what's is done in nuttx/arch/risc-v/include/types.h. The final goal is to remove CONFIG_ARCH_RV32 from Kconfig.

File: arch/risc-v/src/common/riscv_assert.c [hunk 2]
```diff
@@ -71,9 +79,9 @@
  * Name: riscv_stackdump
  ****************************************************************************/
 
-static void riscv_stackdump(uint64_t sp, uintptr_t stack_top)
+static void riscv_stackdump(uintptr_t sp, uintptr_t stack_top)
 {
-  uint64_t stack;
+  uintptr_t stack;
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_assert.c:84]: need update %016" PRIx64 " at line 93

File: arch/risc-v/src/common/riscv_assert.c [hunk 3]
```diff
@@ -98,35 +106,35 @@ static inline void riscv_registerdump(volatile uintptr_t *regs)
...
-  _alert("EPC:%016" PRIx64 "\n", regs[REG_EPC]);
-  _alert("A0:%016" PRIx64 " A1:%01" PRIx64 "6 A2:%016" PRIx64
-         " A3:%016" PRIx64 "\n",
+  _alert("EPC:%0" PRIxREG "\n", regs[REG_EPC]);
+  _alert("A0:%0" PRIxREG " A1:%0" PRIxREG "A2:%0" PRIxREG
+         " A3:%0" PRIxREG "\n",
          regs[REG_A0], regs[REG_A1], regs[REG_A2], regs[REG_A3]);
-  _alert("A4:%016" PRIx64 " A5:%016" PRIx64 "A6:%016" PRIx64
-         " A7:%016" PRIx64 "\n",
+  _alert("A4:%0" PRIxREG " A5:%0" PRIxREG "A6:%0" PRIxREG
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_assert.c:113]: let's move %0 into PRIxREG?

File: arch/risc-v/src/common/riscv_assert.c (hunk 1 above)
Comment (no1wudi) [reply arch/risc-v/src/common/riscv_assert.c:63]: OK

File: arch/risc-v/src/common/riscv_assert.c (hunk 2 above)
Comment (no1wudi) [reply arch/risc-v/src/common/riscv_assert.c:84]: Done

File: arch/risc-v/src/common/riscv_assert.c (hunk 3 above)
Comment (no1wudi) [reply arch/risc-v/src/common/riscv_assert.c]: Done

File: arch/risc-v/src/common/riscv_assert.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_assert.c]: OK.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**